src/collectors/pe/decoder.cpp

```diff
@@ -296,33 +297,105 @@ void pe_decoder<_Bits>::get_imports(boost::filesystem::path const &module_path,
+            logging::log(logging::error) << "ERROR : import name for module " << module_path
                 << " at offset " << std::hex << dll_name_off << " is empty." << std::endl;
             continue;
         }
 
         //names are case unsensitive on windows!
         std::transform(dll_name.begin(), dll_name.end(), dll_name.begin(), ::tolower);
 
+	boost::filesystem::path full_path;
         // try to resolve full module path
         if (!find_module_path(module_path, dll_name)){
             logging::log(logging::warning) << "couldn't find full path for module " << dll_name
                 << " imported by module " << module_path << std::endl;
-        }
+		full_path = Env::root() / "./" / dll_name;
+		logging::log(logging::warning) << "dll name:" << full_path << "\n";
+        }else{
 
         //push module into imports (even if we couldn't find its full path).
-        boost::filesystem::path full_path = Env::root() / dll_name;
+        	full_path = module_path;// Env::root() / dll_name;
+	}
         imports.insert(full_path);
 
+
     } while (::memcmp(&imp_desc, &imp_end, sizeof(PeImportDescriptor)));
+
+
+	uint32_t delay_imports_rva =
+        _pe_data.nt_headers().OptionalHeader.DataDirectory[PeDataDirectory::kEntryDelayImport]
+        .VirtualAddress;
+    uint32_t delay_imports_len =
+        _pe_data.nt_headers().OptionalHeader.DataDirectory[PeDataDirectory::kEntryDelayImport]
+        .Size;
+
+    if (delay_imports_rva == 0 || delay_imports_len == 0){
+        return;
+    }
+
+    uint32_t delay_imports_off;
+    if (!convert_rva_to_offset(delay_imports_rva, delay_imports_off))
+        throw std::runtime_error("pe_decoder::get_delay_imports: bad convert_rva_to_offset");
+
+
+    int step = sizeof(_pe_data.nt_headers().OptionalHeader.ImageBase);
+    uint32_t name_off=1;
+    while(name_off!=0){
+	name_off=0;
+	PeImageDelayImport delay_imports_dir;
+	if (!_pe_data.read(delay_imports_off, sizeof(PeImageDelayImport), reinterpret_cast<char*>(&delay_imports_dir))) {
+		throw std::runtime_error("pe_decoder::get_delay_imports: couldn't read delay_imports directory");
+	}
+
+	if(delay_imports_dir.szName == 0){
+		return;
+	}
+
+	if (!convert_rva_to_offset(delay_imports_dir.szName, name_off)){
+		return;
+	}
+
+	//TODO : associate name + ordinal + func pointer !
+	std::string dep_name;
+	if (!_pe_data.read_line(name_off, dep_name)){
+	    //continue;
+	}else{
+		logging::log(logging::info) << "find delay_import dependencie: " << dep_name <<"\n";
+	}
+
+
+	std::transform(dep_name.begin(), dep_name.end(), dep_name.begin(), ::tolower);
+
+
+	boost::filesystem::path full_path;
+        // try to resolve full module path
+        if (!find_module_path(module_path, dep_name)){
+            logging::log(logging::warning) << "couldn't find full path for module " << dep_name
+                << " imported by module " << module_path << std::endl;
+		full_path = Env::root() / "./" / dep_name;
```

Comment:
this single ``"./" / `` breaks our validation. Why is it there?

Reply:
That's only for the lib not found. If it's found later in an other path, it's changed. That's why at the end, we can find the lib not reached './lib.dll'

Reply:
ok, why not. But in the end I have this path in the data abse: '/./truc.dll' and that's not canonical...